File: mercado_backend/src/main/java/com/mercado/pagos/servicios/service/PagoService.java
package com.mercado.pagos.servicios.service;

import com.mercado.pagos.servicios.dto.request.PagoRequestDTO;
import com.mercado.pagos.servicios.dto.response.PageResponseDTO;
import com.mercado.pagos.servicios.dto.response.PagoResponseDTO;

import java.time.LocalDateTime;

public interface PagoService {

    PagoResponseDTO registrarPago(PagoRequestDTO requestDTO);

    PagoResponseDTO obtenerPagoPorId(Long id);

    PagoResponseDTO obtenerPagoPorDeuda(Long idDeuda);

    PageResponseDTO<PagoResponseDTO> listarPagos(
            String codigoPago,
            String codigoDeuda,
            LocalDateTime inicio,
            LocalDateTime fin,
            int page,
            int size
    );

    PagoResponseDTO anularPago(Long id, String motivo);

}

File: mercado_frontend/src/app/features/pagos/services/pago.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/services/api.service';
import { PageResponseDTO, PagoRequestDTO, PagoResponseDTO } from '../models/pago.models';

@Injectable({
  providedIn: 'root'
})
export class PagoService {
  private readonly api = inject(ApiService);
  private readonly basePath = '/pagos';

  listar(
    page: number,
    size: number,
    filtros?: {
      codigoPago?: string;
      codigoDeuda?: string;
      inicio?: string;
      fin?: string;
    }
  ): Observable<PageResponseDTO<PagoResponseDTO>> {
    return this.api.get<PageResponseDTO<PagoResponseDTO>>(this.basePath, {
      page,
      size,
      codigoPago: filtros?.codigoPago,
      codigoDeuda: filtros?.codigoDeuda,
      inicio: filtros?.inicio,
      fin: filtros?.fin
    });
  }

  registrar(request: PagoRequestDTO): Observable<PagoResponseDTO> {
    return this.api.post<PagoResponseDTO, PagoRequestDTO>(this.basePath, request);
  }

  obtenerPorId(id: number): Observable<PagoResponseDTO> {
    return this.api.get<PagoResponseDTO>(`${this.basePath}/${id}`);
  }

  obtenerPorDeuda(idDeuda: number): Observable<PagoResponseDTO> {
    return this.api.get<PagoResponseDTO>(`${this.basePath}/deuda/${idDeuda}`);
  }

  anular(id: number, motivo: string): Observable<PagoResponseDTO> {
    return this.api.patch<PagoResponseDTO, null>(
      `${this.basePath}/${id}/anular?motivo=${encodeURIComponent(motivo)}`,
      null
    );
  }
}
